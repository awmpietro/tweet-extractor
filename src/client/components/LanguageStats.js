import React from 'react';
import { Table, CardTitle } from 'reactstrap';
import PropTypes from 'prop-types';

const LanguageStats = ({stats}) => {

    const renderLanguages = () => {
        const languages = []
        for(let i in stats.languages){
            languages.push(
            <tr key={i}>
                <td>{i}</td>
                <td>{stats.languages[i]}</td>
            </tr>
            )
        }
        return languages;
    }

    return(
        <>
        <CardTitle><h4>Tweets by language</h4></CardTitle>
        <Table bordered striped>
            <thead>
            <tr>
                <th>Language</th>
                <th>Quantity</th>
            </tr>
            </thead>
            <tbody>
            {renderLanguages()}
            </tbody>
        </Table>
        </>
    )
}

LanguageStats.propTypes = {
    stats: PropTypes.object
};

export default LanguageStats;