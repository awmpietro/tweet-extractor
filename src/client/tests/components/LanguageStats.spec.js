import React from 'react';
import LanguageStats from '../../components/LanguageStats';
import renderer from 'react-test-renderer';

describe('LanguageStats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<LanguageStats stats={{}} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});