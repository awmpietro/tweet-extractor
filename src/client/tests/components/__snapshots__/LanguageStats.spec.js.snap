// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageStats snapshot renders 1`] = `
Array [
  <div
    className="card-title"
  >
    <h4>
      Tweets by language
    </h4>
  </div>,
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th>
          Language
        </th>
        <th>
          Quantity
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
